testing-tools/src/testnet_controllers/strom/strom_internals.rs

```diff
@@ -1,123 +1,73 @@
-    contract_payloads::angstrom::{AngstromPoolConfigStore, UniswapAngstromRegistry},
-    pair_with_price::PairsWithPrice,
-    primitive::{PoolId as AngstromPoolId, UniswapPoolRegistry},
-    sol_bindings::testnet::TestnetHub
+    contract_payloads::angstrom::AngstromPoolConfigStore, pair_with_price::PairsWithPrice,
+    primitive::UniswapPoolRegistry, sol_bindings::testnet::TestnetHub
 };
-use consensus::{AngstromValidator, ConsensusManager, ManagerNetworkDeps, Signer};
-use futures::StreamExt;
+use consensus::{AngstromValidator, ConsensusManager};
+use futures::{StreamExt, TryStreamExt};
 use jsonrpsee::server::ServerBuilder;
-use matching_engine::cfmm::uniswap::{
-    pool::EnhancedUniswapPool, pool_data_loader::DataLoader, pool_manager::UniswapPoolManager,
-    pool_providers::canonical_state_adapter::CanonicalStateAdapter
-};
+use matching_engine::configure_uniswap_manager;
 use order_pool::{order_storage::OrderStorage, PoolConfig};
-use reth_provider::{CanonStateNotifications, CanonStateSubscriptions};
+use reth_provider::CanonStateSubscriptions;
 use reth_tasks::TokioTaskExecutor;
 use secp256k1::SecretKey;
+use tokio_stream::wrappers::BroadcastStream;
 use validation::{
     order::state::{pools::AngstromPoolsTracker, token_pricing::TokenPriceGenerator},
     validator::ValidationClient
 };
 
 use crate::{
     anvil_state_provider::{
-        state_provider_factory::{RpcStateProviderFactory, RpcStateProviderFactoryWrapper},
-        utils::StromContractInstance,
-        AnvilEthDataCleanser
+        utils::StromContractInstance, AnvilEthDataCleanser, AnvilStateProvider,
+        AnvilStateProviderWrapper
     },
-    contracts::deploy_contract_and_create_pool,
-    network::TestnetConsensusFuture,
     testnet_controllers::AngstromTestnetConfig,
-    types::SendingStromHandles,
+    types::{initial_state::InitialTestnetState, SendingStromHandles},
     validation::TestOrderValidator
 };
 
 pub struct AngstromTestnetNodeInternals {
     pub rpc_port:         u64,
-    pub state_provider:   RpcStateProviderFactoryWrapper,
+    pub state_provider:   AnvilStateProviderWrapper,
     pub order_storage:    Arc<OrderStorage>,
     pub pool_handle:      PoolHandle,
     pub tx_strom_handles: SendingStromHandles,
     pub testnet_hub:      StromContractInstance,
-    pub validator:        TestOrderValidator<RpcStateProviderFactory>,
-    _consensus:           TestnetConsensusFuture<PubSubFrontend>,
-    _consensus_running:   Arc<AtomicBool>
+    pub validator:        TestOrderValidator<AnvilStateProvider>
 }
 
 impl AngstromTestnetNodeInternals {
     pub async fn new(
         testnet_node_id: u64,
         strom_handles: StromHandles,
         strom_network_handle: StromNetworkHandle,
-        secret_key: SecretKey,
+        _secret_key: SecretKey,
         config: AngstromTestnetConfig,
-        initial_validators: Vec<AngstromValidator>
-    ) -> eyre::Result<Self> {
+        _initial_validators: Vec<AngstromValidator>,
```

Comment:
if its not used, delete it

Reply:
it's going to be, i commented out a part of the consensus that uses it